src/python/pants/backend/jvm/tasks/jvm_compile/jvm_classpath_publisher.py

```diff
@@ -16,7 +16,13 @@ class RuntimeClasspathPublisher(Task):
   def register_options(cls, register):
     super().register_options(register)
     register('--manifest-jar-only', type=bool, default=False,
+             removal_version='1.25.0.dev2',
+             removal_hint='Use --manifest-jar instead, which respects --internal-classpath-only!',
              help='Only export classpath in a manifest jar.')
+    register('--manifest-jar', type=bool, default=False,
+             help='Export classpath in a manifest jar instead of symlinks in the dist dir.')
```

Comment:
Unless the behavior changes, I'd lean toward keep the option name as is.

Reply:
I wasn't sure how best to avoid breaking existing use cases with the same option, see https://github.com/pantsbuild/pants/pull/8606#discussion_r344458599. I think it makes sense not to change the option name here -- I think one way to resolve this might be to change the default of `--internal-classpath-only` to False, because I feel like providing *more* jars is likely to not break too many workflows (and we can perhaps have a deprecation warning or something to warn people that this change occurred, and how to fix it). I will remove the new `--manifest-jar` option and see if defaulting `--internal-classpath-only=False` can be done gracefully.